### PR TITLE
Allow for a cromwell jar to be passed in

### PR DESCRIPTION
**Pull request by geoffjentry**

### CODE REVIEW

File: test_cromwell.sh [hunk 1]
```diff
@@ -11,49 +11,48 @@ trap "shutdown" EXIT
...
+    r) RUN_DIR=$OPTARG
+       mkdir -p "${RUN_DIR}"
+       ;;
+    c) CONFIG_STRING="-Dconfig.file=$OPTARG"
 	   ;;
-        c) CONFIG_STRING="-Dconfig.file=$OPTARG"
+    p) PARALLELISM_FACTOR=$OPTARG
 	   ;;
-        p) PARALLELISM_FACTOR=$OPTARG
+	j) CROMWELL_JAR=$OPTARG
```
Comment (cjllanwarne) [inline test_cromwell.sh]: curious indentation

Comment (geoffjentry) [inline test_cromwell.sh]: I thought this was an artifact of the diff as I don't see it in my intellij but it seems to be in the raw view here as well

Comment (geoffjentry) [inline test_cromwell.sh]: I do however see it in a terminal. Will fix. IntelliJ is teh suck

File: test_cromwell.sh [hunk 2]
```diff
@@ -11,47 +11,46 @@ trap "shutdown" EXIT
...
     case "$option" in
 	h) echo "$usage"
 	   exit
 	   ;;
 	b) CROMWELL_BRANCH=$OPTARG
 	   ;;
         r) RUN_DIR=$OPTARG
-	   mkdir -p "${RUN_DIR}"
-	   ;;
+           mkdir -p "${RUN_DIR}"
```
Comment (cjllanwarne) [inline test_cromwell.sh]: Indentation still off w.r.t `h)` and `b)`

File: test_cromwell.sh [hunk 3]
```diff
@@ -11,51 +11,50 @@ trap "shutdown" EXIT
 
 set -e
 
-EXIT_CODE=31
+EXIT_CODE=1
```
Comment (Horneth) [inline test_cromwell.sh:14]: Just curious, why this change ? Also what is it ? The exit code of this script ?

Comment (geoffjentry) [inline test_cromwell.sh:14]: Cuz 31 I left in by accident earlier on when I was making sure I was seeing this particular exit code.

This is the default exit code, it gets flipped to 0 at the very end. The reason is that I have the EXIT trap to clean things up at the end but that trap function can't know the exit code that led to the exit.

**Comment by Horneth** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/75/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: test_cromwell.sh [hunk 4]
```diff
@@ -11,51 +11,50 @@ trap "shutdown" EXIT
...
     -c    If supplied, the config file to pass to Cromwell
     -p    Number of simultaneous tests to run. Defaults to 3
 "
 
 INITIAL_DIR=$(pwd)
-
-PARALLELISM_FACTOR=3
-CROMWELL_BRANCH="develop"
 RUN_DIR=$(pwd)
-CONFIG_STRING=""
```
Comment (cjllanwarne) [inline test_cromwell.sh:34]: Was this removed on purpose? Seems useful (from my "don't forget to initialise to null" days)

File: test_cromwell.sh [hunk 5]
```diff
@@ -11,51 +11,50 @@ trap "shutdown" EXIT
...
+            exit
+            ;;
+        b) CROMWELL_BRANCH=$OPTARG
+            ;;
         r) RUN_DIR=$OPTARG
-	   mkdir -p "${RUN_DIR}"
-	   ;;
+            mkdir -p "${RUN_DIR}"
+            ;;
         c) CONFIG_STRING="-Dconfig.file=$OPTARG"
```
Comment (cjllanwarne) [inline test_cromwell.sh]: I agree with `#92` since that would cause all args to be lumped together... OTOH, this one probably should have quotes around `$OPTARGS`?

But OT**O**H, I don't really care

File: test_cromwell.sh (hunk 4 above)
Comment (geoffjentry) [inline test_cromwell.sh:34]: I realized it was superfluous. If it doesn't exist it drops in nothing as well

File: test_cromwell.sh (hunk 5 above)
Comment (geoffjentry) [inline test_cromwell.sh]: $OPTARGS is already in quotes. Perhaps it could use braces. The linter didn't complain.

File: test_cromwell.sh [hunk 6]
```diff
@@ -67,37 +66,54 @@ CENTAUR_LOG=${LOG_DIR}/centaur.log
...
 
-# Build and run cromwell
-if [ ! -d "cromwell" ]; then
-    echo "Cloning Cromwell"
-    git clone https://github.com/broadinstitute/cromwell.git
-else
-    echo "Using existing Cromwell directory"
+if [[ -n ${CROMWELL_BRANCH} ]]; then
+    if [[ -n ${CROMWELL_JAR} ]]; then
+        echo "Do not specify both a branch and a jar"
```
Comment (cjllanwarne) [inline test_cromwell.sh]: `>&2`

File: test_cromwell.sh [hunk 7]
```diff
@@ -67,37 +66,54 @@ CENTAUR_LOG=${LOG_DIR}/centaur.log
...
+echo "Starting Cromwell, jar is ${CROMWELL_JAR}"
+# Adding quotes on CONFIG_STRING caused bad things to happen. Sorry shellcheck
+java ${CONFIG_STRING} -jar "${CROMWELL_JAR}" server >> "${CROMWELL_LOG}" 2>&1 &
 
 # Build and run centaur
-cd ..
+cd "${RUN_DIR}"
 
-if [ ! -d "centaur" ]; then
+if [[ ! -d "centaur" ]]; then
```
Comment (cjllanwarne) [inline test_cromwell.sh:97]: TOL: Does shellcheck not complain about not checking whether there's a file there instead?

File: test_cromwell.sh (hunk 5 above)
Comment (cjllanwarne) [inline test_cromwell.sh]: What I meant was, you want the `"`s to appear in the result string, so that `CONFIG_STRING` would end up looking like `-Dconfig.file="hello world.txt"` instead of `-Dconfig.file=hello world.txt`

**Comment by cjllanwarne** [comment]
I wouldn't have been too offended if you'd just self-👍'd this

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/75/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: test_cromwell.sh (hunk 5 above)
Comment (geoffjentry) [inline test_cromwell.sh]: Ah I see. I just assume people aren't monsters using spaces in file names.

It turns out that @kshakir chided me earlier for doing that line wrong anyways. Will change it.

File: test_cromwell.sh (hunk 7 above)
Comment (geoffjentry) [inline test_cromwell.sh:97]: No, and it's a good point although beyond the level of handhold-y I originally intended. I'll put this under the "things maybe I'll clean up in the near future"
